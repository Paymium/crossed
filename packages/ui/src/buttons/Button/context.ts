/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createContext } from 'react';
import { ButtonProps } from './types';

export const buttonContext = createContext<
  Pick<ButtonProps, 'variant' | 'size' | 'disabled' | 'error'> & {
    state?: {
      active?: boolean;
      hover?: boolean;
    };
    textId: string;
    // eslint-disable-next-line no-unused-vars
    setTextId: (p: string) => void;
  }
>({} as any);
