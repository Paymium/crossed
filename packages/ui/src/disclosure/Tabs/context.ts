/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createScope } from '@crossed/core';
import { ScrollView } from 'react-native-gesture-handler';
import { SharedValue } from 'react-native-reanimated';

export type TabsContext = {
  value: string | number;
  setValue: (_value: string | number) => void;
  id: string;
  variant?: 'underline' | 'rounded';
  size?: 'sm' | 'md' | 'lg';
  listTabRef: React.MutableRefObject<ScrollView>;
  indicator: { left: SharedValue<number>; width: SharedValue<number> };
  scroll: SharedValue<number>;
  shouldShow: boolean;
  setShow: React.Dispatch<React.SetStateAction<boolean>>;
  widthLayout: SharedValue<number>;
};

export type TriggerContext = {
  disabled?: boolean;
  hover?: boolean;
  selected?: boolean;
};

export const createContext = () => {
  return {
    tabsContext: createScope<TabsContext>({
      variant: 'rounded',
    } as TabsContext),
    triggerContext: createScope<TriggerContext>({}),
  };
};
