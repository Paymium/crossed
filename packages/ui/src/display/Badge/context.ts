/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createScope } from '@crossed/core';
import { Context } from './type';

export const [Provider, useContext] = createScope<Context>({} as Context);
export const [ProviderGroup, useContextGroup] = createScope<Context>(
  {} as Context
);
