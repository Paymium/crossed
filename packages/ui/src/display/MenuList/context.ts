/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createScope } from '@crossed/core';
import { Context } from './types';

export const [Provider, useProviderContext] = createScope<Context>(
  {} as Context
);
