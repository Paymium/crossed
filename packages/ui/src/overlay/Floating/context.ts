/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createScope } from '@crossed/core';

export type FloatingContext = {
  open: boolean;
  onClose: () => void;
  onOpen: () => void;
  visibilityHidden?: boolean;
};

export const [FloatingProvider, useFloatingContext] =
  createScope<FloatingContext>({} as FloatingContext);
