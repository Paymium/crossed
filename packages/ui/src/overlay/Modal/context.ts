/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createContext } from 'react';

export type VariantSize = { size: 'sm' | 'md' | 'lg' };

export const localContext = createContext<VariantSize>({ size: 'md' });
