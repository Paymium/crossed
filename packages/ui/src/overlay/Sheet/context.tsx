/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { createContext, ForwardedRef, RefObject, useContext } from 'react';
import type { SharedValue } from 'react-native-reanimated';
import { ActionSheetRef } from '@crossed/sheet';

export type SheetContext = {
  open: boolean;
  setOpen: (_value: boolean) => void;
  dismissOnOverlayPress?: boolean;
  hideHandle?: boolean;
  isMove: SharedValue<boolean>;
  height: SharedValue<number>;
  onClose: () => void;
  snapInitialHeight: SharedValue<number>;
  offset: number;
  ref: ForwardedRef<ActionSheetRef>;
  actionSheetRef: RefObject<ActionSheetRef>;
};

export const sheetContext = createContext<SheetContext>({} as SheetContext);

export const useSheetContext = () => useContext(sheetContext);
