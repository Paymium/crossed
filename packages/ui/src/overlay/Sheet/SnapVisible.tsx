/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import { composeEventHandlers } from '@crossed/core';
import { YBox, type YBoxProps } from '../../layout/YBox';
import { useCallback } from 'react';
import { type LayoutChangeEvent } from 'react-native';
import { useSheetContext } from './context';

export const SnapVisible = (props: YBoxProps) => {
  const { snapInitialHeight, offset } = useSheetContext();
  const onLayout = useCallback(
    ({ nativeEvent }: LayoutChangeEvent) => {
      const OFFSET = nativeEvent.layout.height + 40 + offset;

      if (Math.floor(snapInitialHeight.value) !== Math.floor(OFFSET))
        snapInitialHeight.value = OFFSET;
    },
    [snapInitialHeight, offset]
  );

  return (
    <YBox
      {...props}
      onLayout={composeEventHandlers(onLayout, props.onLayout)}
    />
  );
};
