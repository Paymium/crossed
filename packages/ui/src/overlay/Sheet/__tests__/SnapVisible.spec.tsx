/**
 * Copyright (c) Paymium.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root of this projects source tree.
 */

import React from 'react';
import { SnapVisible } from '../SnapVisible';
import { render, screen } from '@crossed/test';
import { sheetContext } from '../context';
import { expect, describe, test } from '@jest/globals';
import { PortalProvider } from '@gorhom/portal';

describe('SnapVisible', () => {
  test('snapshot', async () => {
    const snapInitialHeight = { value: 20 };
    const offset = 0;
    render(
      <PortalProvider>
        <sheetContext.Provider
          value={
            {
              snapInitialHeight,
              offset,
            } as any
          }
        >
          <SnapVisible testID="test" />
        </sheetContext.Provider>
      </PortalProvider>
    );
    expect(await screen.findByTestId('test')).toMatchSnapshot();
  });
});
