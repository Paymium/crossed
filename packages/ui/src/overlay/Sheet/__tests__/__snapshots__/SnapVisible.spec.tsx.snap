// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapVisible snapshot 1`] = `
<div
  class="css-view-175oi2r border-style-[solid] border-width-[0px] align-items-[stretch] background-color-[rgba(0,-0,-0,-0)] display-[flex] flex-basis-[auto] flex-direction-[column] flex-shrink-[0] list-style-[none] margin-[0px] min-height-[0px] min-width-[0px] padding-[0px] position-[relative] text-decoration-[none] z-index-[0] align-self-[stretch] box-sizing-[border-box]"
  data-testid="test"
/>
`;
